Source/WebCore/dom/NameNodeList.cpp

```diff
@@ -52,7 +53,7 @@ NameNodeList::~NameNodeList()
 
 bool NameNodeList::elementMatches(Element& element) const
 {
-    return element.isHTMLElement() && element.getNameAttribute() == m_name;
+    return (is<HTMLElement>(element) || element.document().quirks().needsGetElementsByNameQuirk()) && element.getNameAttribute() == m_name;
```

Comment:
I don't think we want to be checking the quirk value for every element like this.
We should probably cache the value in a member variable so that we don't do a function call & two indirect loads for every element.

Reply:
Done. Not entirely sure if the const_cast is avoidable. But I hope that we can remove this code within a year.